nixos/modules/services/web-apps/pretalx.nix

```diff
@@ -0,0 +1,406 @@
+{ config
+, options
+, lib
+, pkgs
+, ...
+}:
+
+let
+  cfg = config.services.pretalx;
+  opt = options.services.pretalx;
+
+  inherit (lib)
+    escapeShellArgs
+    literalExpression
+    mdDoc
+    mkDefault
+    mkEnableOption
+    mkIf
+    mkPackageOptionMD
+    mkOption
+    optionalString
+    recursiveUpdate
+    types;
+
+  format = pkgs.formats.ini {};
+
+  extras = cfg.package.optional-dependencies.redis
+    ++ lib.optionals (cfg.settings.database.backend == "mysql") cfg.package.optional-dependencies.mysql
+    ++ lib.optionals (cfg.settings.database.backend == "postgresql") cfg.package.optional-dependencies.postgres;
+
+  PYTHONPATH = "${cfg.package.PYTHONPATH}:${cfg.package.python.pkgs.makePythonPath extras}";
```

Comment:
Could you construct an env using `withPackages` instead? I don't know if it's a problem in this particular instance, but we don't want spawned child processes to inherit `PYTHONPATH`.

Reply:
Would this be enough for you @adisbladis https://github.com/NixOS/nixpkgs/commit/e9ada499eba5706a8ad2320cb1ed1dcd5350a613 ?